include/session.hh

```diff
@@ -27,63 +27,104 @@ public:
 
     ~session();
 
+    prepared_query prepare(const std::string &query);
+
+    const CassPrepared *prepare_raw(const std::string& query);
+
     future execute_async(const statement &statement);
+    future execute_async(const statement &&statement);
 
     // Convenience functions to avoid creating scd_statement manually for simple queries
     future execute_async(const std::string &query);
+    future execute_async(const std::string &&query);
 
-    future execute_async(const batch_query& query);
+    future execute_async(const batch_query &query);
+    future execute_async(const batch_query &&query);
 
-    future execute_async(const prepared_query& query);
+    future execute_async(const prepared_query &query);
+    future execute_async(const prepared_query &&query);
 
     template<typename... Args,
         typename = typename std::enable_if<0 != sizeof...(Args)>::type>
     future execute_async(const std::string &query, Args... args) {
         return execute_async(scmd::statement(query, sizeof...(args)).bind(args...));
     }
+    template<typename... Args,
+        typename = typename std::enable_if<0 != sizeof...(Args)>::type>
+    future execute_async(const std::string &&query, Args... args) {
+        return execute_async(scmd::statement(query, sizeof...(args)).bind(args...));
```

Comment:
I think the query should be moved here further to the statement constructor, if possible.